homeassistant/components/zwave_js/sensor.py

```diff
@@ -736,24 +736,11 @@ def __init__(
             alternate_value_name=self.info.primary_value.property_name,
             additional_info=[self.info.primary_value.property_key_name],
         )
-
-    @property
-    def options(self) -> list[str] | None:
-        """Return options for enum sensor."""
-        if self.device_class == SensorDeviceClass.ENUM:
-            return list(self.info.primary_value.metadata.states.values())
-        return None
-
-    @property  # type: ignore[override]
-    # fget is used which is not compatible with cached_property
-    # mypy also doesn't know about fget: https://github.com/python/mypy/issues/6185
-    def device_class(self) -> SensorDeviceClass | None:
-        """Return sensor device class."""
-        if (device_class := super().device_class) is not None:
-            return device_class
-        if self.info.primary_value.metadata.states:
-            return SensorDeviceClass.ENUM
-        return None
+        if (device_class := self.entity_description.device_class) is not None:
+            self._attr_device_class = device_class
+        elif self.info.primary_value.metadata.states:
+            self._attr_device_class = SensorDeviceClass.ENUM
+            self._attr_options = list(info.primary_value.metadata.states.values())
```

Comment:
I think the `device_class` in `ZWaveBaseEntity` is going to win here so it won't see these

Reply:
Actually I think I'm looking on the wrong line since this inherits from `ZwaveSensor`

Reply:
so `ZWaveListSensor` -> `ZwaveSensor` -> `ZWaveBaseEntity`

`ZwaveSensor` does device `device_class` so it wins out of this. 

I'm tired though so I might be reading it wrong

Reply:
It is a pitty the code coverage is lacking. I suppose an additional test is need.

Reply:
The inheritance logic here is really tricky

Reply:
I think it can be simplified to 

```diff
diff --git a/homeassistant/components/zwave_js/sensor.py b/homeassistant/components/zwave_js/sensor.py
index fd4cd78c086..257644e4b34 100644
--- a/homeassistant/components/zwave_js/sensor.py
+++ b/homeassistant/components/zwave_js/sensor.py
@@ -645,13 +645,6 @@ class ZwaveSensor(ZWaveBaseEntity, SensorEntity):
             return None
         return str(self.info.primary_value.metadata.unit)
 
-    @property  # type: ignore[override]
-    # fget is used in the child classes which is not compatible with cached_property
-    # mypy also doesn't know about fget: https://github.com/python/mypy/issues/6185
-    def device_class(self) -> SensorDeviceClass | None:
-        """Return device class of sensor."""
-        return super().device_class
-
 
 class ZWaveNumericSensor(ZwaveSensor):
     """Representation of a Z-Wave Numeric sensor."""
@@ -736,9 +729,7 @@ class ZWaveListSensor(ZwaveSensor):
             alternate_value_name=self.info.primary_value.property_name,
             additional_info=[self.info.primary_value.property_key_name],
         )
-        if (device_class := self.entity_description.device_class) is not None:
-            self._attr_device_class = device_class
-        elif self.info.primary_value.metadata.states:
+        if self.info.primary_value.metadata.states:
             self._attr_device_class = SensorDeviceClass.ENUM
             self._attr_options = list(info.primary_value.metadata.states.values())
 
@@ -777,8 +768,7 @@ class ZWaveConfigParameterSensor(ZWaveListSensor):
             additional_info=[property_key_name] if property_key_name else None,
         )
         if (
-            self.entity_description.device_class is None
-            and not hasattr(self, "_attr_device_class")
+            not self.device_class
             and self._primary_value.configuration_value_type
             == ConfigurationValueType.ENUMERATED
         ):

```

Reply:
No that won't work either

Reply:
```diff
diff --git a/homeassistant/components/zwave_js/sensor.py b/homeassistant/components/zwave_js/sensor.py
index fd4cd78c086..6059eeee524 100644
--- a/homeassistant/components/zwave_js/sensor.py
+++ b/homeassistant/components/zwave_js/sensor.py
@@ -645,13 +645,6 @@ class ZwaveSensor(ZWaveBaseEntity, SensorEntity):
             return None
         return str(self.info.primary_value.metadata.unit)
 
-    @property  # type: ignore[override]
-    # fget is used in the child classes which is not compatible with cached_property
-    # mypy also doesn't know about fget: https://github.com/python/mypy/issues/6185
-    def device_class(self) -> SensorDeviceClass | None:
-        """Return device class of sensor."""
-        return super().device_class
-
 
 class ZWaveNumericSensor(ZwaveSensor):
     """Representation of a Z-Wave Numeric sensor."""
@@ -736,9 +729,7 @@ class ZWaveListSensor(ZwaveSensor):
             alternate_value_name=self.info.primary_value.property_name,
             additional_info=[self.info.primary_value.property_key_name],
         )
-        if (device_class := self.entity_description.device_class) is not None:
-            self._attr_device_class = device_class
-        elif self.info.primary_value.metadata.states:
+        if self.info.primary_value.metadata.states:
             self._attr_device_class = SensorDeviceClass.ENUM
             self._attr_options = list(info.primary_value.metadata.states.values())
 


```

Reply:
I don't have any `ZWaveConfigParameterSensor` or `ZWaveListSensor` entities so no way to test this and it looks like the tests don't cover it.

I did test the PR as-is on my august lock and the other ones seem fine

Reply:
It is indeed not needed to check the super device class. Feel free to add the commit.